### ISSUE TITLE
[Bug] Method parameters cannot be announced by Screen Reader

### ISSUE THREAD

**Issue by yanshudan** [eclipse-platform/eclipse.platform.ui#2795]
Let's make sure issue is not already fixed in latest builds first.

- [x] I verified I can reproduce this issue against [latest Integration Build of Eclipse SDK](https://download.eclipse.org/eclipse/downloads/)

## Steps to reproduce

From a fresh installation and clean workspace:
Install Screen Reader [NVDA](https://www.nvaccess.org/download/)
Enable `Show method paramters`
![Image](https://github.com/user-attachments/assets/949ddd9b-dd29-4520-85c8-9c528a5a94c2)

I tried

input some method call with parameters. Method parameters are shown in the editor like this:

![Image](https://github.com/user-attachments/assets/aaeac822-ff25-459d-a215-ea5d8e5cbe7d)

move the caret to this line, screen reader will announce the content of this line

I expected: The announced text should contain the method paramters, here is the word "delimiter" and "elements"

But got: the method parameter names are skipped by Screen Reader

## Tested under this environment:
* OS & version: Win11
* Eclipse IDE/Platform version 2024-09

## Community

- [x] I understand reporting an issue to this OSS project does not mandate anyone to fix it. Other contributors may consider the issue, or not, at their own convenience. The most efficient way to get it fixed is that I fix it myself and contribute it back as a good quality patch to the project.

**Comment by mickaelistria** [comment]
Thanks for reporting. Accessibility support for code minings is indeed non-existent. There would be ways to improve those (such as making the accessibility listener for SWT StyledText request information as to what is drawn by the code mining), but one would actually need to design an accessibility API for annotations (similarly to supporting `alt` attribute on HTML images).
Would you be interested in trying to contribute that?

**Comment by BeckerWdf** [comment]
I think it's quite hard to add this from a user point of view. So you have to make clear to the user of the screensread which text is actually in the editor and which not not "really" there but only as code mining.

In our product (ABAP Development Tools) we use the code mining as a means to show code completion proposals and for this we added an separate setting which when turned on does show the completions in a dialog (which is read by the screenreaders) instead of code minings. But his is specific to this "code completion" use case.

The "I show additional information in the editor" use case is a completely different use case.
